### PR TITLE
allow passing a http.RoundTripper to NewHeaderClient

### PR DESCRIPTION
**Pull request by mpuncel**

### CODE REVIEW

File: pkg/pods/pod.go [hunk 1]
```diff
@@ -186,10 +186,6 @@ func (pod *Pod) buildRunitServices(launchables []hoist.Launchable) error {
 		for _, executable := range executables {
 			sbTemplate.AddEntry(executable.Service.Name, executable.Exec)
 		}
```
Comment (bwester) [inline pkg/pods/pod.go]: Removing this error message seems unrelated to the TLS configuration change. Can you put it in its own commit?

Comment (mpuncel) [inline pkg/pods/pod.go]: sure, although it's not really a change since it's unreachable code

File: pkg/util/net/client.go [hunk 2]
```diff
@@ -22,14 +23,22 @@ func (ht headerTransport) RoundTrip(r *http.Request) (*http.Response, error) {
 // NewHeaderClient returns an http.Client. It has the same behavior as
 // DefaultClient, except that the key-value pairs in extras are added as headers
 // on every request.
-func NewHeaderClient(extras map[string]string) *http.Client {
+func NewHeaderClient(extras map[string]string, tlsConfig *tls.Config) *http.Client {
```
Comment (bwester) [inline pkg/util/net/client.go]: Exposing only the TLS config seems rather arbitrary. Instead of accepting at TLS config and building the inner transport from that, what about accepting an `http.RoundTripper` and letting the caller decide how to initialize it?  Or perhaps a `*http.RoundTripper`, so that `nil` can be provided to get a sensible default?

Comment (mpuncel) [inline pkg/util/net/client.go]: sounds good

Comment (mpuncel) [inline pkg/util/net/client.go]: http.RoundTripper is an interface so `nil` is valid for it

File: bin/p2-inspect/main.go [hunk 3]
```diff
@@ -51,7 +51,7 @@ func main() {
 	opts := kp.Options{
 		Address: *consulUrl,
 		Token:   *consulToken,
-		Client:  net.NewHeaderClient(*headers),
+		Client:  net.NewHeaderClient(*headers, nil),
```
Comment (anthonybishopric) [inline bin/p2-inspect/main.go]: I'd feel more comfortable explicitly passing `http.DefaultTransport`, but I won't press that point.

File: pkg/pods/pod.go (hunk 1 above)
Comment (anthonybishopric) [inline pkg/pods/pod.go]: I would like to do this in a separate commit only because I think we need to replace this message with something else. Ideally we provide information to the service healthcheck about the uptime of services and the PIDs they're running as.

Comment (mpuncel) [inline pkg/pods/pod.go]: by separate commit do you mean separate PR? Also this if statement is totally superfluous and will never execute if you read the code above it

File: bin/p2-inspect/main.go (hunk 3 above)
Comment (mpuncel) [inline bin/p2-inspect/main.go]: that sounds good
